src/main/java/pages/LoginPage.java

```diff
@@ -0,0 +1,58 @@
+package pages;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+
+public class LoginPage {
+
+    private By loginBtn = By.xpath("//div[@class='HeadBanner-ButtonsWrapper']/a[2]");
+    private By submitBtn = By.xpath("//button[@type='submit']");
+    private By loginField = By.xpath("//input[@name='login']");
+    private By passwordField = By.xpath("//input[@name='passwd']");
+    private By sigInBtn = By.xpath("//button[@id='passp:sign-in']");
+    private By composeButton = By.xpath("//a[@href='#compose']");
+
+
+    WebDriver driver;
+
+    public LoginPage(WebDriver driver) {
+        this.driver = driver;
+    }
+
+    public boolean login(String login, String password) {
+        WebDriverWait wait = new WebDriverWait(driver, 10);
+
+        wait.until(ExpectedConditions.visibilityOfAllElementsLocatedBy(loginBtn));
```

Comment:
too many waiters. I think you don't need almost all of them. You can read about implicit wait

Reply:
fixed